### DISCUSSION TITLE
tsh login doesn't work when put Teleport cluster behind Cloudfront

### DISCUSSION THREAD

**Discussion by smeeklai** [gravitational/teleport#16624] [Q&A] [answered]
I installed `teleport-cluster` helm chart and can get Teleport up and running. It was working fine when I point my record to the LB directly. However, I would like to make Teleport only accessible from our VPN IP. I couldn't do that via a security group since Teleport use `nlb` as the LB type. So I decided to create a Cloudfront and use WAF to control IP access. Then I found a problem when performing `tsh login`. I got an timeout error saying:
```
ERROR: failed to authenticate with proxy teleport.production-internal.opn.dev:3023
dial tcp 65.9.181.113:3023: connect: operation timed out
```
But if I log in to Teleport via web UI, it works fine.

**Comment by webvictim** [comment]
`tsh login` does indeed require a TCP connection to port 3023 to work correctly.

You might be able to use TLS routing/multiplexing to send all traffic over port 443, as long as you aren't terminating TLS at the load balancer: https://goteleport.com/docs/architecture/tls-routing/

**Comment by smeeklai** [comment, marked as answer]
Thank you. So Cloudfront can't be use since I believe it does terminate TLS. I continued to research more and found that I could use below annotation for NLB to whitelist certain IPs to fulfill my goal.
```
service.beta.kubernetes.io/load-balancer-source-ranges
```
